In [1]:
## Muestra de el archivo inicial con scrapers
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Escuelas complex
df_schools_complex = pd.read_csv('./data/thesis_tutor_7078_clean_v1.csv')
df_schools_complex

Unnamed: 0  index  thesis_year  \
0              0      1         2009   
1              1     95  20-Jun-2011   
2              2     22  31-May-2011   
3              3     24  31-May-2011   
4              4     25  31-May-2011   
...          ...    ...          ...   
7073        7098   8190  12-Nov-2020   
7074        7099   8192  12-Nov-2020   
7075        7100   8193  18-Nov-2020   
7076        7101   8196  18-Nov-2020   
7077        7102   8211  12-Mar-2021   

                                           thesis_title  \
0     Estudio de sobretensiones por maniobra debido ...   
1     Estudio de la influencia de las condiciones de...   
2     Efecto de los rayos gamma (g) sobre el comport...   
3     Determinación de registros pseudo sónicos a pa...   
4     Propuesta de un sistema de administración de i...   
...                                                 ...   
7073  Magnitud y estructura del abastecimiento exter...   
7074  Evaluación del Impacto de Cambios de Cobertura...   
7075  Efecto de la sombra sobre la concentración de ...   
7076  Evaluación del comportamiento ecofisiológico d...   
7077  La industria musical surcoreana, recepción y a...   

                                    thesis_author       size  \
0                         De Gregorio S., Luis T.    6.77 MB   
1     Hernández F., Adamarlys; Jiménez G., Harold    2.46 MB   
2                        Rodríguez G., Marlene C.    1.62 MB   
3                              Urbina P., Eros R.   11.62 MB   
4                             Sánchez M., José R.  562.96 kB   
...                                           ...        ...   
7073                        González V., Jorge C.    1.09 MB   
7074                            Rojas N., Ladu G.    3.29 MB   
7075                         Herrera B., Mauro D.  706.17 kB   
7076                              Alfonzo, Miguel    2.14 MB   
7077                  Utrera Ávila, Joselin Marly    2.33 MB   

                                 thesis_link  \
0     http://saber.ucv.ve/handle/10872/17712   
1       http://saber.ucv.ve/handle/10872/160   
2        http://saber.ucv.ve/handle/10872/42   
3        http://saber.ucv.ve/handle/10872/53   
4        http://saber.ucv.ve/handle/10872/45   
...                                      ...   
7073  http://saber.ucv.ve/handle/10872/20882   
7074  http://saber.ucv.ve/handle/10872/20885   
7075  http://saber.ucv.ve/handle/10872/20888   
7076  http://saber.ucv.ve/handle/10872/20891   
7077  http://saber.ucv.ve/handle/10872/20974   

                                               pdf_link  resumen  \
0     http://saber.ucv.ve/bitstream/10872/17712/1/TE...      NaN   
1     http://saber.ucv.ve/bitstream/10872/160/1/TEG.pdf      NaN   
2     http://saber.ucv.ve/bitstream/10872/42/1/Tesis...      NaN   
3     http://saber.ucv.ve/bitstream/10872/53/1/TRABA...      NaN   
4     http://saber.ucv.ve/bitstream/10872/45/1/TRABA...      NaN   
...                                                 ...      ...   
7073  http://saber.ucv.ve/bitstream/10872/20882/1/Ma...      NaN   
7074  http://saber.ucv.ve/bitstream/10872/20885/1/Tr...      NaN   
7075  http://saber.ucv.ve/bitstream/10872/20888/1/Te...      NaN   
7076  http://saber.ucv.ve/bitstream/10872/20891/1/TE...      NaN   
7077  http://saber.ucv.ve/bitstream/10872/20974/1/TF...      NaN   

                         path  exist  isPDF  isScan  \
0        thesis_pdf_all/1.pdf   True   True   False   
1       thesis_pdf_all/95.pdf   True   True   False   
2       thesis_pdf_all/22.pdf   True   True   False   
3       thesis_pdf_all/24.pdf   True   True   False   
4       thesis_pdf_all/25.pdf   True   True   False   
...                       ...    ...    ...     ...   
7073  thesis_pdf_all/8190.pdf   True   True   False   
7074  thesis_pdf_all/8192.pdf   True   True   False   
7075  thesis_pdf_all/8193.pdf   True   True   False   
7076  thesis_pdf_all/8196.pdf   True   True   False   
7077  thesis_pdf_all/8211.pdf   True   True   Fa

In [2]:
len(df_schools_complex)

7078

In [3]:
df_schools_complex = df_schools_complex.iloc[: , 1:]
df_schools_complex

index  thesis_year                                       thesis_title  \
0         1         2009  Estudio de sobretensiones por maniobra debido ...   
1        95  20-Jun-2011  Estudio de la influencia de las condiciones de...   
2        22  31-May-2011  Efecto de los rayos gamma (g) sobre el comport...   
3        24  31-May-2011  Determinación de registros pseudo sónicos a pa...   
4        25  31-May-2011  Propuesta de un sistema de administración de i...   
...     ...          ...                                                ...   
7073   8190  12-Nov-2020  Magnitud y estructura del abastecimiento exter...   
7074   8192  12-Nov-2020  Evaluación del Impacto de Cambios de Cobertura...   
7075   8193  18-Nov-2020  Efecto de la sombra sobre la concentración de ...   
7076   8196  18-Nov-2020  Evaluación del comportamiento ecofisiológico d...   
7077   8211  12-Mar-2021  La industria musical surcoreana, recepción y a...   

                                    thesis_author       size  \
0                         De Gregorio S., Luis T.    6.77 MB   
1     Hernández F., Adamarlys; Jiménez G., Harold    2.46 MB   
2                        Rodríguez G., Marlene C.    1.62 MB   
3                              Urbina P., Eros R.   11.62 MB   
4                             Sánchez M., José R.  562.96 kB   
...                                           ...        ...   
7073                        González V., Jorge C.    1.09 MB   
7074                            Rojas N., Ladu G.    3.29 MB   
7075                         Herrera B., Mauro D.  706.17 kB   
7076                              Alfonzo, Miguel    2.14 MB   
7077                  Utrera Ávila, Joselin Marly    2.33 MB   

                                 thesis_link  \
0     http://saber.ucv.ve/handle/10872/17712   
1       http://saber.ucv.ve/handle/10872/160   
2        http://saber.ucv.ve/handle/10872/42   
3        http://saber.ucv.ve/handle/10872/53   
4        http://saber.ucv.ve/handle/10872/45   
...                                      ...   
7073  http://saber.ucv.ve/handle/10872/20882   
7074  http://saber.ucv.ve/handle/10872/20885   
7075  http://saber.ucv.ve/handle/10872/20888   
7076  http://saber.ucv.ve/handle/10872/20891   
7077  http://saber.ucv.ve/handle/10872/20974   

                                               pdf_link  resumen  \
0     http://saber.ucv.ve/bitstream/10872/17712/1/TE...      NaN   
1     http://saber.ucv.ve/bitstream/10872/160/1/TEG.pdf      NaN   
2     http://saber.ucv.ve/bitstream/10872/42/1/Tesis...      NaN   
3     http://saber.ucv.ve/bitstream/10872/53/1/TRABA...      NaN   
4     http://saber.ucv.ve/bitstream/10872/45/1/TRABA...      NaN   
...                                                 ...      ...   
7073  http://saber.ucv.ve/bitstream/10872/20882/1/Ma...      NaN   
7074  http://saber.ucv.ve/bitstream/10872/20885/1/Tr...      NaN   
7075  http://saber.ucv.ve/bitstream/10872/20888/1/Te...      NaN   
7076  http://saber.ucv.ve/bitstream/10872/20891/1/TE...      NaN   
7077  http://saber.ucv.ve/bitstream/10872/20974/1/TF...      NaN   

                         path  exist  isPDF  isScan  \
0        thesis_pdf_all/1.pdf   True   True   False   
1       thesis_pdf_all/95.pdf   True   True   False   
2       thesis_pdf_all/22.pdf   True   True   False   
3       thesis_pdf_all/24.pdf   True   True   False   
4       thesis_pdf_all/25.pdf   True   True   False   
...                       ...    ...    ...     ...   
7073  thesis_pdf_all/8190.pdf   True   True   False   
7074  thesis_pdf_all/8192.pdf   True   True   False   
7075  thesis_pdf_all/8193.pdf   True   True   False   
7076  thesis_pdf_all/8196.pdf   True   True   False   
7077  thesis_pdf_all/8211.pdf   True   True   False   

                               school_simple  \
0            escuela de ingenieria electrica   
1              escuela de ingenieria quimica   
2              escuela de ingenieria quimica   
3     escuela de geologia, minas y geofis

In [4]:
df_schools_complex.to_csv('./data/thesis_tutor_7078_clean_v2.csv', index=False)

In [3]:
# Select random rows
df_schools_sample365 = df_schools_complex.sample(n = 365, random_state = 2)
df_schools_sample365

Unnamed: 0  index  thesis_year  \
1116        1117   1638  22-Feb-2013   
4597        4615   5419  27-Mar-2017   
4668        4686   5506  28-Mar-2017   
2719        2725   3364   2-Jun-2015   
3339        3349   4021   7-Dec-2015   
...          ...    ...          ...   
6709        6734   7830  20-Oct-2019   
5535        5557   6529  31-Jan-2018   
2576        2580   3184  11-May-2015   
3077        3085   3641  22-Sep-2015   
2553        2556   3188  11-May-2015   

                                           thesis_title  \
1116  "Creciendo Saludable" Diseño de una revista de...   
4597  Factibilidad del uso de la técnica multicompon...   
4668  Análisis geomecánico y estabilidad de los estr...   
2719  Teatro costumbrista\r\nTres Autores Tres Época...   
3339  Monitoreo de los sistemas de energía y factore...   
...                                                 ...   
6709  Relación entre el lenguaje no verbal y la viol...   
5535  Desarrollo de norma técnica para los enlaces m...   
2576  Efecto de microondas en el tamaño de gotas de ...   
3077  Condiciones edáficas de los herbazales halófit...   
2553  Composición química y disponibilidad ambiental...   

                   thesis_author       size  \
1116          Brito, Danielys D.    1.82 MB   
4597          Ramos A., Simón E.    7.26 MB   
4668  Manuel Rico, Vicente Johan   28.68 MB   
2719              Núñez, Lisbeth    3.03 MB   
3339      Freijanes P., Dario J.     2.6 MB   
...                          ...        ...   
6709           Hernández, Daniel  492.86 kB   
5535           Goa H., Dayana A.    1.53 MB   
2576         Prado S., Andrea E.    4.69 MB   
3077              Marrero, María    3.39 MB   
2553          González, Anamelia    2.21 MB   

                                 thesis_link  \
1116   http://saber.ucv.ve/handle/10872/2780   
4597  http://saber.ucv.ve/handle/10872/15406   
4668  http://saber.ucv.ve/handle/10872/15454   
2719   http://saber.ucv.ve/handle/10872/9604   
3339  http://saber.ucv.ve/handle/10872/13000   
...                                      ...   
6709  http://saber.ucv.ve/handle/10872/20219   
5535  http://saber.ucv.ve/handle/10872/17688   
2576   http://saber.ucv.ve/handle/10872/9174   
3077  http://saber.ucv.ve/handle/10872/11895   
2553   http://saber.ucv.ve/handle/10872/9166   

                                               pdf_link  resumen  \
1116  http://saber.ucv.ve/bitstream/10872/2780/1/Tes...      NaN   
4597  http://saber.ucv.ve/bitstream/10872/15406/1/TE...      NaN   
4668  http://saber.ucv.ve/bitstream/10872/15454/1/Tr...      NaN   
2719  http://saber.ucv.ve/bitstream/10872/9604/1/Tes...      NaN   
3339  http://saber.ucv.ve/bitstream/10872/13000/1/TR...      NaN   
...                                                 ...      ...   
6709  http://saber.ucv.ve/bitstream/10872/20219/1/CL...      NaN   
5535  http://saber.ucv.ve/bitstream/10872/17688/1/Te...      NaN   
2576  http://saber.ucv.ve/bitstream/10872/9174/1/TEG...      NaN   
3077  http://saber.ucv.ve/bitstream/10872/11895/1/Te...      NaN   
2553  http://saber.ucv.ve/bitstream/10872/9166/1/Tes...      NaN   

                         path  exist  isPDF  isScan  \
1116  thesis_pdf_all/1638.pdf   True   True   False   
4597  thesis_pdf_all/5419.pdf   True   True   False   
4668  thesis_pdf_all/5506.pdf   True   True   False   
2719  thesis_pdf_all/3364.pdf   True   True   False   
3339  thesis_pdf_all/4021.pdf   True   True   False   
...                       ...    ...    ...     ...   
6709  thesis_pdf_all/7830.pdf   True   True   False   
5535  thesis_pdf_all/6529.pdf   True   True   False   
2576  thesis_pdf_all/3184.pdf   True   True   False   
3077  thesis_pdf_all/3641.pdf   True   True   False   
2553  thesis_pdf_all/3188.pdf   True   True   False   

                               school_simple  \
1116          escuela de comunicacion social   
4597  escuela de geologia, minas y geofisica   
4668  escuela de geologia, minas y geofisica   
2719

In [1120]:
df_schools_sample = df_schools_sample365.iloc[: , 1:]
df_schools_sample

index  thesis_year                                       thesis_title  \
1116   1638  22-Feb-2013  "Creciendo Saludable" Diseño de una revista de...   
4597   5419  27-Mar-2017  Factibilidad del uso de la técnica multicompon...   
4668   5506  28-Mar-2017  Análisis geomecánico y estabilidad de los estr...   
2719   3364   2-Jun-2015  Teatro costumbrista\r\nTres Autores Tres Época...   
3339   4021   7-Dec-2015  Monitoreo de los sistemas de energía y factore...   
642    1025  25-Jul-2012            La Hoyada, Espacio para la Comunicación   
2660   3313  25-May-2015  Estudio geoquímico exploratorio de los sedimen...   
3144   3780   1-Oct-2015  Evaluación sismorresistente del comportamiento...   
887     962  18-Jul-2012  Almodóvar: La Madre que lo Parió  Significació...   
877    1230  14-Nov-2012  Medidas de bioseguridad utilizadas por el pers...   
4021   4939  23-Feb-2017  Geología y geomecánica del macizo rocoso San J...   
844    1254  22-Nov-2012  Propuesta de un programa para la prevención de...   
4071   4757   3-Feb-2017  Propuesta de un programa de capacitación integ...   
5810   6963  31-May-2018  Desarrollo de un sistema mecánico para equipos...   
4242   5111   7-Mar-2017  Estudio del efecto de un metilendioxi-flavonoi...   
4607   5430  27-Mar-2017  Estudio de formación de los complejos ternario...   
2151   2364  10-Jun-2014  Análisis exergénico de un equipo de acondicion...   
2264   2808  17-Nov-2014  Una aplicación Web para el manejo de datos de ...   
1409   2198   3-May-2014  Formación en valores y Disciplina escolar en c...   
6925   8038  14-Jan-2020  Desarrollo de una solución de inteligencia de ...   
3490   4230  18-Feb-2016  Digestibilidad in vitro y aprovechamiento por ...   
1998   2573   7-Jul-2014  Estudio experimental del comportamiento térmic...   
207     292  17-Oct-2011  Propuesta de rediseño ergonómico del quirófano...   
4253   5030   2-Mar-2017  Aplicación de la técnica de Inpainting en un e...   
1112   1634  21-Feb-2013  Expresiones musicales venezolanas. Tres micros...   
463     824  20-Jun-2012  Modelos epistemológicos humanismo y empirismo ...   
3273   3985  30-Nov-2015  Selección de un sistema de gestión de red para...   
6011   7021   7-Jun-2018  Catálogo de la sección fotográfica Teresa Carr...   
4171   4955  23-Feb-2017  Estudio físico en Ia disociación de estructura...   
1089   1505  30-Jan-2013  Clima organizacional dada la satisfacción labo...   
2420   2986   6-Feb-2015  Propuesta de un plan operativo comunal para el...   
2929   3532  13-Jul-2015  Estudio de modelos de rigidez lateral equivale...   
6275   7315   4-Dec-2018  Estudio del efecto del tiempo de envejecimient...   
6822   7964  10-Dec-2019  Calidad de suelo y micotrofía en tres agroecos...   
3125   3756  30-Sep-2015  Evaluación de diferentes métodos de extracción...   
6251   7322   4-Dec-2018  Caracterización geomecánica del yacimiento cre...   
1455   2004  11-Dec-2013  Simulación en estado estacionario de un hemodi...   
3469   3829  16-Oct-2015                   Fórmula de Inversión de Lagrange   
194     248  13-Oct-2011  Estudio cortical en el área centro-norte y nor...   
6663   7910  20-Nov-2019  "Desde mis zapatos: Una mirada desde la divers...   
6995   7809  11-Oct-2019  Desempeño laboral del psicólogo (a) en el Cent...   
1590   2035   8-Jan-2014  Características métricas en maxilares inferior...   
1614   2118   7-Feb-2014  Mujer, profetiza y líder. Aproximación antropo...   
5172   6044  18-Oct-2017  Riesgo cardiotóxico de la infusión de hojas fr...   
6562   7294  16-Nov-2018  Veinte años de la Educación Básica en Venezuel...   
7059   8108  19-Jun-2020  Diseño de un plan de gestión que integre las t...   
4237   5197  13-Mar-2017  Desarrollo de un sistema de información para l...   
1252   1664  27-Feb-2013  Diseño de un sistema de gestión de calidad par...   
4730   5580  31-Mar-2017  Estudio de  la formación de los complejos de n...   
1469   1926  11-Oct-2013  Diseño de fluidos viscoelástic

In [57]:
pd.set_option('display.max_rows', None)
df_schools_sample365_sorted = df_schools_sample365[['index','school_complex','thesis_tutor', 'thesis_author']].sort_values(by=['index'])
df_schools_sample365_sorted

index                                     school_complex  \
186      35             Escuela de Geología, Minas y Geofísica   
110      45                  Escuela de Ingeniería de Petroleo   
37       57             Escuela de Geología, Minas y Geofísica   
103      71                     Escuela de Ingeniería Mecanica   
62       88                        Escuela de Ingeniería Civil   
20      103  Escuela de Ingeniería Metalúrgica y Ciencia de...   
88      115             Escuela de Geología, Minas y Geofísica   
145     177                              Escuela de Enfermería   
194     248             Escuela de Geología, Minas y Geofísica   
129     270                              Escuela de Enfermería   
130     271                              Escuela de Enfermería   
207     292                     Escuela de Ingeniería Mecanica   
273     308                              Escuela de Enfermería   
256     350                    Escuela de Ingeniería Eléctrica   
379     645                              Escuela de Enfermería   
380     646                            Escuela de Antropología   
81      684                              Escuela de Enfermería   
367     696                     Escuela de Comunicación Social   
368     698                               Escuela de Filosofía   
391     724                     Escuela de Comunicación Social   
460     821                              Escuela de Enfermería   
463     824                              Escuela de Enfermería   
495     826                              Escuela de Enfermería   
472     835                      Escuela de Ingeniería Química   
489     853  Escuela de Ingeniería Metalúrgica y Ciencia de...   
500     896                              Escuela de Enfermería   
604     923                              Escuela de Enfermería   
558     934                     Escuela de Comunicación Social   
562     938                     Escuela de Comunicación Social   
565     941                     Escuela de Comunicación Social   
586     952                     Escuela de Comunicación Social   
887     962                     Escuela de Comunicación Social   
595     974                     Escuela de Comunicación Social   
584     975                     Escuela de Comunicación Social   
642    1025                     Escuela de Comunicación Social   
649    1033                     Escuela de Comunicación Social   
671    1059                     Escuela de Comunicación Social   
681    1069                     Escuela de Comunicación Social   
740    1133                                   Escuela de Artes   
829    1175                               Escuela de Filosofía   
814    1219  Escuela de Ingeniería Metalúrgica y Ciencia de...   
849    1228                              Escuela de Enfermería   
877    1230                              Escuela de Enfermería   
837    1247                               Escuela de Educación   
838    1248                               Escuela de Educación   
844    1254                              Escuela de Enfermería   
799    1267                              Escuela de Enfermería   
866    1279  Escuela de Ingeniería Metalúrgica y Ciencia de...   
889    1351                     Escuela de Comunicación Social   
942    1384                     Escuela de Ingeniería Mecanica   
1104   1495                      Escuela de Ingeniería Química   
1089   1505                              Escuela de Enfermería   
1019   1519                              Escuela de Enfermería   
1142   1541                     Escuela de Comunicación Social   
1160   1563                  Escuela de Ingeniería de Petroleo   
1287   1581                     Escuela de Comunicación Social   
1289   1584                     Escuela de Comunicación Social   
1167   1598                  Escuela de Ingeniería de Petroleo   
1202   1608                     Escuela de Comunicación Social   
1209   1615                     Escuela de Com

In [1114]:
df_schools_sample365_sorted.drop(index=df_schools_sample365_sorted.index[0], axis=0, inplace=True)
df_schools_sample365_sorted

index         school_complex         thesis_tutor  \
7077   8211  Escuela de Sociología  Juan Francisco Sans   

                    thesis_author  
7077  Utrera Ávila, Joselin Marly

In [1115]:
next_number = df_schools_sample365_sorted['index'].to_list()[0]
next_number

8211

In [1116]:
new_list_st = append_row(new_list_st, next_number, 1, 1)
new_list_st

[(35, 1, 1),
 (45, 1, 1),
 (57, 1, 1),
 (71, 1, 0),
 (88, 1, 1),
 (103, 1, 1),
 (115, 1, 1),
 (177, 1, 1),
 (248, 1, 1),
 (270, 1, 1),
 (271, 1, 1),
 (292, 1, 1),
 (308, 1, 1),
 (350, 1, 0),
 (645, 1, 0),
 (646, 1, 1),
 (684, 1, 1),
 (696, 1, 1),
 (698, 1, 1),
 (724, 1, 1),
 (821, 1, 1),
 (824, 1, 1),
 (826, 1, 0),
 (835, 1, 1),
 (853, 1, 1),
 (896, 1, 0),
 (923, 1, 1),
 (934, 1, 1),
 (938, 1, 1),
 (941, 1, 1),
 (952, 1, 1),
 (962, 1, 1),
 (974, 1, 1),
 (975, 1, 1),
 (1025, 1, 1),
 (1033, 1, 1),
 (1059, 1, 1),
 (1069, 1, 1),
 (1133, 1, 1),
 (1175, 1, 1),
 (1219, 1, 1),
 (1228, 1, 0),
 (1230, 1, 1),
 (1247, 1, 0),
 (1248, 1, 1),
 (1254, 1, 1),
 (1267, 1, 0),
 (1279, 1, 1),
 (1351, 1, 1),
 (1384, 1, 1),
 (1495, 1, 1),
 (1505, 1, 1),
 (1519, 1, 0),
 (1541, 1, 1),
 (1563, 1, 1),
 (1581, 1, 1),
 (1584, 1, 1),
 (1598, 1, 1),
 (1608, 1, 1),
 (1615, 1, 1),
 (1634, 1, 1),
 (1638, 1, 1),
 (1664, 1, 1),
 (1675, 1, 1),
 (1693, 1, 1),
 (1695, 1, 1),
 (1702, 1, 1),
 (1707, 1, 1),
 (1765, 1, 1),
 (17

In [42]:
### (index, school, tutor)
array_school_true = [(35, 1, 1), (45, 1, 1), (57, 1, 1,), (71, 1, 0), (88, 1, 1), (103, 1, 1),
                    (115, 1, 1), (177, 1, 1)]
# el nombre del tutor si esta presente en el resultado es tomado como resutlado correcto incluso si tiene lic antes, u otras palabras despues como presentado.
# si tiene dos tutores y identifica a por lo menos 1 correctamente se toma como correcto.
## tesis 1942 tiene el tutor como autor en saber.ucv
## 2166 tutor con caracas luego del nombre
## 2657 tutor con presentado
## 2819 no hay tutor en ninguna parte
## 2908 tutor con caracas luego del nombre
## 3158 tutor con trabajo luego del nombre
## 3188 no hay tutor
## 3222 tutor con caracas al lado
## 3474 no es una tesis es un resumen
## 3825 tutor con caracas
## 3829 tutor con caracas
## 3904 tutor con caracas
## 3977 tutor con tutor al lado
## profesor guia en vez de decir tutor
## 4549 profesor  on trabajo
## 4787 tutor con caracas
## 4919 tiene algo de basura al lado
## 4955 quien sabe
## 5030 caracas despues de tutor
## 5036 estan escaneados
## 5135 basura al lado
## 5650 en otra parte el tutor
## 5725 caracas tutor
## 5910 cotutro luego del nombre
## 6008 Caracas
## 6531 Es tutor indutrial
## 7012 tutora despues del nombre
## 7081 dice papa
## 7837 hugo chavez aunque esta en el titulo
## 7963 Br en tutor
## 8087 trabajo luego del nombre

In [1117]:
len(new_list_st)

365

In [14]:
def append_row(list_array, index, school, tutor):
    list_array_aux = list_array.copy()
    list_array_aux.append((index, school, tutor))
    return list_array_aux

In [1074]:
new_list_st.pop()

(7861, 1, 1)

In [1118]:
df_st = pd.DataFrame(new_list_st, columns=['index','school_true','tutor_true'])
df_st

index  school_true  tutor_true
0       35            1           1
1       45            1           1
2       57            1           1
3       71            1           0
4       88            1           1
5      103            1           1
6      115            1           1
7      177            1           1
8      248            1           1
9      270            1           1
10     271            1           1
11     292            1           1
12     308            1           1
13     350            1           0
14     645            1           0
15     646            1           1
16     684            1           1
17     696            1           1
18     698            1           1
19     724            1           1
20     821            1           1
21     824            1           1
22     826            1           0
23     835            1           1
24     853            1           1
25     896            1           0
26     923            1           1
27     934            1           1
28     938            1           1
29     941            1           1
30     952            1           1
31     962            1           1
32     974            1           1
33     975            1           1
34    1025            1           1
35    1033            1           1
36    1059            1           1
37    1069            1           1
38    1133            1           1
39    1175            1           1
40    1219            1           1
41    1228            1           0
42    1230            1           1
43    1247            1           0
44    1248            1           1
45    1254            1           1
46    1267            1           0
47    1279            1           1
48    1351            1           1
49    1384            1           1
50    1495            1           1
51    1505            1           1
52    1519            1           0
53    1541            1           1
54    1563            1           1
55    1581            1           1
56    1584            1           1
57    1598            1           1
58    1608            1           1
59    1615            1           1
60    1634            1           1
61    1638            1           1
62    1664            1           1
63    1675            1           1
64    1693            1           1
65    1695            1           1
66    1702            1           1
67    1707            1           1
68    1765            1           1
69    1776            1           1
70    1793            1           1
71    1810            1           0
72    1843            1           1
73    1870            1           1
74    1883            1           1
75    1904            1           1
76    1926            1           1
77    1938            1           1
78    1942            1           0
79    1954            1           1
80    1976            1           1
81    2004            1           1
82    1986            1           1
83    2004            1           1
84    2009            1           1
85    2035            1           1
86    2060            1           1
87    2097            1           1
88    2118            1           1
89    2166            1           1
90    2184            1           1
91    2185            1           1
92    2198            1           0
93    2200            1           0
94    2208            1           1
95    2211            1           1
96    2221            1           1
97    2255            1           1
98    2308            1           1
99    2336            1           1
100   2348            1           1
101   2350            1           1
102   2364            1           1
103   2378            1           1
104   2383            1           1
105   2406            1           1
106   2407            1           1
107   2435            1           1
108   2447            1           1
109   2573            1           1
110   260

In [1136]:
df_tutor_false = df_st[df_st['tutor_true']==0]
len(df_tutor_false)

34

In [1137]:
df_tutor_true = df_st[df_st['tutor_true']==1]
len(df_tutor_true)

331

In [1128]:
df_school = df_st[df_st['school_true']==1]
len(df_school)

365

In [1123]:
df_st_joined = pd.merge(df_st, df_schools_sample, how='inner')
df_st_joined

index  school_true  tutor_true  thesis_year  \
0       35            1           1   1-Jun-2011   
1       45            1           1   3-Jun-2011   
2       57            1           1   3-Jun-2011   
3       71            1           0   7-Jun-2011   
4       88            1           1  15-Jun-2011   
5      103            1           1  22-Jun-2011   
6      115            1           1  22-Jun-2011   
7      177            1           1  30-Sep-2011   
8      248            1           1  13-Oct-2011   
9      270            1           1  17-Oct-2011   
10     271            1           1  17-Oct-2011   
11     292            1           1  17-Oct-2011   
12     308            1           1  17-Oct-2011   
13     350            1           0  20-Oct-2011   
14     645            1           0  22-May-2012   
15     646            1           1  22-May-2012   
16     684            1           1  25-May-2012   
17     696            1           1  29-May-2012   
18     698            1           1  29-May-2012   
19     724            1           1   4-Jun-2012   
20     821            1           1  20-Jun-2012   
21     824            1           1  20-Jun-2012   
22     826            1           0  20-Jun-2012   
23     835            1           1   2-Jul-2012   
24     853            1           1   2-Jul-2012   
25     896            1           0   9-Jul-2012   
26     923            1           1  11-Jul-2012   
27     934            1           1  16-Jul-2012   
28     938            1           1  16-Jul-2012   
29     941            1           1  16-Jul-2012   
30     952            1           1  16-Jul-2012   
31     962            1           1  18-Jul-2012   
32     974            1           1  18-Jul-2012   
33     975            1           1  18-Jul-2012   
34    1025            1           1  25-Jul-2012   
35    1033            1           1  20-Sep-2012   
36    1059            1           1  21-Sep-2012   
37    1069            1           1  25-Sep-2012   
38    1133            1           1  11-Oct-2012   
39    1175            1           1  31-Oct-2012   
40    1219            1           1  13-Nov-2012   
41    1228            1           0  14-Nov-2012   
42    1230            1           1  14-Nov-2012   
43    1247            1           0  16-Nov-2012   
44    1248            1           1  22-Nov-2012   
45    1254            1           1  22-Nov-2012   
46    1267            1           0  22-Nov-2012   
47    1279            1           1  26-Nov-2012   
48    1351            1           1  10-Jan-2013   
49    1384            1           1  17-Jan-2013   
50    1495            1           1  30-Jan-2013   
51    1505            1           1  30-Jan-2013   
52    1519            1           0  31-Jan-2013   
53    1541            1           1   1-Feb-2013   
54    1563            1           1   5-Feb-2013   
55    1581            1           1  13-Feb-2013   
56    1584            1           1  13-Feb-2013   
57    1598            1           1  14-Feb-2013   
58    1608            1           1  18-Feb-2013   
59    1615            1           1  18-Feb-2013   
60    1634            1           1  21-Feb-2013   
61    1638            1           1  22-Feb-2013   
62    1664            1           1  27-Feb-2013   
63    1675            1           1  28-Feb-2013   
64    1693            1           1  28-Feb-2013   
65    1695            1           1  28-Feb-2013   
66    1702            1           1  12-Mar-2013   
67    1707            1           1  14-Mar-2013   
68    1765            1           1   9-May-2013   
69    1776            1           1   9-May-2013   
70    1793            1           1  10-May-2013   
71    1810            1           0  13-May-2013   
72    1843            1           1  17-May-2013   
73    1870            1           1  30-Sep-2013   
74    1883            1           1   1-Oct-2013   
75    1904            1           1   4-Oct-2013   
7

In [1138]:
len(df_tutor_true)/len(new_list_st)

0.9068493150684932

## Por Tutor
Hay 90.68% de precision (esto es igual a 34 de 365 tesis fueron etiquetadas erronamente), con 95% de confianza y 5% de margen de error podemos afirmar que con 90.68% de las tesis estan correctamente clasificadas por tutor. Esto significa tambien que con 95% de confianza entramos en 85% de precision con las tesis correctamente clasificadas en el peor de los casos.


## Por escuela
Hay 100% de precision de las 365 tesis todas fueron etiquetadas correctamente por escuela. Con 95% de confianza y 5% de margen error que 100% de las tesis estan correctamente clasificadas por carrera.

In [10]:
#277
#1703
#2543
#4127
#5448
#6584
#7627
#8147

array_school_true = [1, 1, 1, 1, 1, 1, 1, 1, 1, 0,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
        1, 1, 1, 1, 1, 1, 1, 0, 1, 1,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
        1, 1, 1, 1, 1, 1, 1, 1
       ]

# 1596 Escribio la escuela de quimica primero que ingenieria de petroleo.
# 5344 Escribio Ingenieria Quimica aparece como Metalurgica??
# 5657 Ingenieria de petroleo escrita correcta...

In [12]:
df_schools_sample68_sorted['true_school'] = array_school_true
df_schools_sample68_sorted

index                                     school_complex  true_school
37      277                              Escuela de Enfermería            1
288     324                      Escuela de Ingeniería Química            1
1475    439             Escuela de Geología, Minas y Geofísica            1
814     893                              Escuela de Enfermería            1
661     966                     Escuela de Comunicación Social            1
497    1011                     Escuela de Comunicación Social            1
681    1240                                   Escuela de Artes            1
1259   1350                     Escuela de Comunicación Social            1
1272   1396                     Escuela de Ingeniería Mecanica            1
5975   1596                      Escuela de Ingeniería Química            0
1290   1703                     Escuela de Ingeniería Mecanica            1
1440   1815                      Escuela de Ingeniería Química            1
1445   1821                      Escuela de Ingeniería Química            1
1447   1823                      Escuela de Ingeniería Química            1
1698   1859             Escuela de Geología, Minas y Geofísica            1
1660   2062                               Escuela de Educación            1
1667   2069                               Escuela de Educación            1
1034   2104                              Escuela de Enfermería            1
1755   2172                              Escuela de Enfermería            1
1903   2435                     Escuela de Ingeniería Mecanica            1
2072   2543                     Escuela de Ingeniería Mecanica            1
2080   2581                     Escuela de Ingeniería Mecanica            1
2218   2646                     Escuela de Ingeniería Mecanica            1
2204   2753                             Escuela de Computación            1
2046   2763                             Escuela de Computación            1
2310   2840                                 Escuela de Química            1
2378   2923                                Escuela de Biología            1
2637   3249                                Escuela de Biología            1
3479   3827                              Escuela de Matemática            1
3455   4043                    Escuela de Ingeniería Eléctrica            1
3528   4127                    Escuela de Ingeniería Eléctrica            1
3734   4372                      Escuela de Ingeniería Química            1
3851   4499                              Escuela de Enfermería            1
3809   4513                              Escuela de Enfermería            1
4260   4999             Escuela de Geología, Minas y Geofísica            1
4493   5293             Escuela de Geología, Minas y Geofísica            1
4537   5313                      Escuela de Ingeniería Química            1
4554   5344  Escuela de Ingeniería Metalúrgica y Ciencia de...            0
4633   5359             Escuela de Geología, Minas y Geofísica            1
4616   5403                             Escuela de Computación            1
4688   5448             Escuela de Geología, Minas y Geofísica            1
4832   5608                                  Escuela de Física            1
4848   5657                  Escuela de Ingeniería de Petroleo            1
4943   5735                             Escuela de Computación            1
5329   6016                                 Escuela de Química            1
5170   6026                                Escuela de Biología            1
5805   6335                               Escuela de Educación            1
5420   6426                      Escuela de Ingeniería Química            1
6872   6485                               Escuela de Educación            1
6899   6575                    Escuela de Ingeniería Eléctrica            1
6881   6584                             Escuela de Computación            1
6848   6675                               Escuela de E

In [18]:
df_schools_sample68_sorted['true_school'].value_counts()

1    66
0     2
Name: true_school, dtype: int64

In [19]:
df_schools_sample68_sorted['true_school'].value_counts()[1]/len(df_schools_sample68_sorted)

0.9705882352941176

Hay 97.05% de precision, con 90% de confianza y 10% de margen de error podemos afirmar que 97.05% de las tesis estan correctamente clasificadas por carrera. Esto significa tambien que con 90% de confianza estamos entre 87.05% de las tesis correctamete clasificadas en el peor de los casos. Sigue siendo un numero lo suficientemete grande.

En el entrenaimento se usaron 20 tesis, 5 de validacion y 5 de pureba. En total 30 tesis de computacion de un total de 524 tesis de computacion clasificadas. Serian 0.5% de las tesis fueron tomadas de forma arbitraria. Se realizaron 5 preguntas de cada una de las tesis y 6 en una tesis de validacion de id 5727. Todas estas tesis se verificaron manualmente que fueran de computacion.  